deepchecks/recommender/ranking.py

```diff
@@ -7,705 +7,423 @@
-
-    if return_matrix:
-        out_mat = np.zeros((n, n))
-        for i in range(n):
-            for j in range(i):
-                out_mat[i][j] = np.dot(a_norm[i], a_norm[j])
-        out_mat = out_mat + out_mat.T
-        np.fill_diagonal(out_mat, 1)
-        return out_mat
-    else:
-        out = np.zeros((len(perm), 1))
-        for i in range(len(perm)):
-            out[i] = np.dot(a_norm[perm[i][0]], a_norm[perm[i][1]])
-        return out
-
-
-# Metrics implemented internally
-# =================================================================================================
-
-def inner_diversity(item_features):
-    """Calculate the diversity of a recommendation list based on an internally provided similarity matrix.
+    Compute the Recall@k score for a recommendation.
 
     Args:
-        item_features (array-like): An N x M array of item features.
+        relevant_items (Union[List[int], np.array, None]): The set of relevant items for the user.
+        recommendation (Union[List, np.array, None]): The recommendation list to be evaluated.
+        k (int): The number of top items to consider.
 
     Returns:
-        diversity (float): The diversity of the recommendation list.
-    """
-    # calculate the cosine similarity between the items
-    similarity = cosine_pairwise_distance(item_features, return_matrix=True)
-    # calculate the diversity
-    return 1 - np.mean(similarity[np.triu_indices(len(similarity), k=1)])
-
-
-def diversity(recommendation: List, item_features: pd.DataFrame) -> float:
+        float: The Recall@k score for the recommendation.
     """
-    Calculate the diversity of a recommendation list based on an externally provided similarity matrix
-
-    Args:
-        recommendation (array-like): An N x 1 array of ordered items.
-        item_features (array-like): An N x M array of item features.
-    Returns:
-        diversity (float): The diversity of the recommendation list.
-    """
-
-    # select only the features of the recommended items
-    item_features = item_features.loc[recommendation, :]
-
-    return inner_diversity(item_features.values.astype(np.float32))
-
-# The following metrics where copied from https://github.com/AstraZeneca/rexmex
-# Under the Apache License Version 2.0, January 2004 http://www.apache.org/licenses/.
-# Citation: Rozemberczki, B., Nilsson, S., Hoyt, C. T., & Edwards, G. RexMex (Version 0.1.0) [Computer software].
-# https://github.com/AstraZeneca/rexmex
-## =================================================================================================
+    if relevant_items is None or len(relevant_items) == 0:
+        return 0.0
+    if recommendation is None or len(recommendation) == 0:
+        return 0.0
+    relevant_set = set(relevant_items)
+    rec_set = set(recommendation[:k])
+    hits = list((i in relevant_set) for i in recommendation if i in rec_set)
+    score = np.cumsum(hits) / (1 + np.arange(len(hits)))
```

Comment:
For recall, using the **set()** function helps to remove any duplicates, allowing us to skip the step of checking whether a recommendation has already been made and speed up computation. However, it is crucial to address and raise awareness of any duplicates that may arise during the recommendation process. This should be considered mandatory and included in the Sanity Check  of the recommender to ensure its reliability.

In general, when using a hybrid model that employs heuristic logic (such as incorporating both popularity and recency), it is recommended to apply the **set()** function to every recommendation to eliminate any potential duplicate candidates.

It's also important to note, that we shouldn't use **set()** for the argument :<recommandations> of precision@K because we care about the order. I already corrected this in my actual fork.